llvm/test/tools/llvm-dwarfdump/debug-names-local-tu-bad-die.yaml

```diff
@@ -0,0 +1,116 @@
+# RUN: yaml2obj %s -o - | not llvm-dwarfdump -verify - | FileCheck %s
```

Comment:
Judging by the strings in this file, you've produced this by running (meta:P) clang over some file, and then tweaking the output. Would it be possible to make the same modifications on the intermediate assembly file instead?

I see there is already a precedent for yaml input files in this folder, so I wouldn't say this is a hard requirement, but I think it'd much easier to see what's going on here by looking at the (annotated) assembly than at a raw hex dump. (At this point, I've gotten pretty good at reading dwarf from assembly, but I'm not l33t enough to do it from hex :P)

Reply:
This is a linked executable. It doesn't work on .o files with multiple DWARF units. I tried to make one and ended up with:
```
Verifying /Users/gclayton/Documents/src/github/Debug/test/tools/llvm-dwarfdump/X86/Output/debug-names-verify-tu-bad-die.s.tmp:  file format elf64-x86-64
Verifying .debug_abbrev...
Verifying .debug_info Unit Header Chain...
Verifying .debug_types Unit Header Chain...
Verifying non-dwo Units...
Verifying unit: 1 / 2
Verifying unit: 2 / 2, "main.cpp"
Verifying dwo Units...
Verifying .debug_line...
warning: .debug_line[0x00000000].prologue.file_names[1] is a duplicate of file_names[0]
Verifying .debug_str_offsets...
Verifying .debug_names...
error: Name Index @ 0x0: Entry @ 0x86: mismatched Tag of DIE @ 0x23: index - DW_TAG_subprogram; debug_info - DW_TAG_structure_type.
error: Name Index @ 0x0: Entry @ 0x86: mismatched Name of DIE @ 0x23: index - main; debug_info - Foo.
error: Aggregated error counts:
error: Name Index contains mismatched Tag of DIE occurred 1 time(s).
error: Name Index contains mismatched name of DIE occurred 1 time(s).
Errors detected.
```

Reply:
And this was for a .o file straight from the compiler. I am guessing llvm-dwarfdump might not be applying relocations, but that is beyond the scope of this patch.

Is there a way to get a .s file for linked executable?

Reply:
There's no way to get a .s version of a linked executable. I guess it's possible there's something wrong with the relocations, though I'm somewhat surprised by that since llvm-dwarfdump is reguarly used (both in tests and in real word) on unlinked files. What's the input you used to reproduce this?